### ISSUE TITLE
What about a standalone (non-grunt) version ?

### ISSUE THREAD

**Issue by MoOx** [filamentgroup/grunticon#134]
It would be really cool to avoid the grunt dependancy.
Obviously I understand a grunt task is a cool thing, but the ability to use it whitout grunt is more cool.
Also it would allow people to make a gulp (or whatever) version.
Don't you think ?

**Comment by bonfish** [comment]
+1 Would be great.

**Comment by suisho** [comment]
+1

**Comment by oskarrough** [comment]
Provided you already have grunticon set up using a gruntfile, you can wrap it with gulp (for whatever reason you might have…) like this:

```js
var gulp = require('gulp');
var exec = require('child_process').exec;

// Runs grunticon directly from grunt
gulp.task('icons', function (cb) {
	exec('grunt grunticon', function (err, stdout, stderr) {
		console.log(stdout);
		console.log(stderr);
		cb(err);
	});
});```

**Comment by jefflembeck** [comment]
Psst. https://github.com/filamentgroup/grunticon-lib

**Comment by jefflembeck** [comment]
Anyway, that's a stand-alone version, it's in beta now. I plan on trying to fold it into this as soon as possible

**Comment by thalesfsp** [comment]
There's some update about a gulpincon? ;)

**Comment by jefflembeck** [comment]
So, at this point we don't intend to write a gulp version of grunticon (this may change, we're all just pretty busy right now), but with a stand-alone version, it should be a pretty simple task for folks to take on.

**Comment by dcalhoun** [comment]
@jefflembeck I am attempting to utilize grunticon-lib in a Gulp project. The current README claims the [listOfFiles parameter accepts a string](https://github.com/filamentgroup/grunticon-lib#listoffiles) for a directory to read. However, it [doesn't appear to accept anything other than an array](https://github.com/filamentgroup/grunticon-lib/blob/cb609ae822f13eececd9d3e28bf614f572ab7a09/lib/grunticon-lib.js#L56-L58) currently. Is reading a directory not implemented currently?

Thanks for the work on this.

**Comment by jefflembeck** [comment]
@dcalhoun good catch. It's not that it's implemented incorrectly as much as I haven't updated the documentation yet. Woops. Sorry.

I strongly recommend using `fs.readdir` or `fs.readdirSync` and passing in the array that gives you.

**Comment by dcalhoun** [comment]
@jefflembeck thanks for the help. I was able to get it up and running.

For anyone who follows that is interested, [here is a Gist of my example Gulp task](https://gist.github.com/dcalhoun/e79ad10d518612d70721).

**Comment by jefflembeck** [comment]
This is closed for 2.2, which has shipped!
